sparql-visualizer/visualizations/graphical/GraphicalVisualizerTest.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { additionalSymbolEncoding, constructGraphicalVisualizerUrl } from './graphicalVisualizer';
+type VisualisationIdentifier = import('./index.types').VisualisationIdentifier;
+
+describe('GraphicalVisualizer', () => {
+    it('should encode the specified query for the HTML format', () => {
+        const endpoint: string = 'https://query.wikidata.org/';
+        const visualisationType: VisualisationIdentifier = 'BubbleChart';
+        const sparqlQuery: string = `SELECT ?count ?venue (SAMPLE(?venue_label_) AS ?venue_label)
+WITH {
+  SELECT (COUNT(?work) as ?count) ?venue WHERE {
+    ?work wdt:P50 wd:Q16733372 .
+    ?work wdt:P1433 ?venue .
+  }
```

Comment:
In den Testfälle in denen es um `constructGraphicalVisualizerUrl` geht möchtest du ja keine anderen Funktionen mit testen.

Also eigentlich gibt es bei dieser Funktion 2 testbare Eigenschaften:
* Die Reihenfolge der URL Komponenten muss  stimmen
* Bestimmte Zeichen müssen escaped werden

Um das zu testen muss man nicht unbedingt echte Queries benutzen, da reicht auch ein Beispielstring wie `const sparqlQuery: string = 'abc%()/&#'`. Dann sind die Tests kürzer, verständlicher  und für dich leichter zu schreiben.

Reply:
Das kannst du für einige weitere Tests auch so anwenden, man muss nicht überall echte Daten haben, es geht nur um die Funktionalität.

Reply:
Ich denke das Testen von korrekten Zeichen escapes ist unsinnig weil, ich somit nur JS encodeURIComponent teste. 

Für den selbst geschriebenen escaper hab ich schon separate test cases. 

Wenn ich alles in einem teste wäre es doch ein Integration-Test.